crates/ty_python_semantic/src/types/class.rs

```diff
@@ -1348,11 +1348,21 @@ impl<'db> ClassLiteral<'db> {
         let class_definition =
             semantic_index(db, self.file(db)).expect_single_definition(class_stmt);
 
-        class_stmt
-            .bases()
-            .iter()
-            .map(|base_node| definition_expression_type(db, class_definition, base_node))
-            .collect()
+        if self.is_known(db, KnownClass::VersionInfo) {
+            let tuple_type = TupleType::new(db, TupleSpec::version_info_spec(db))
+                .expect("sys.version_info tuple spec should always be a valid tuple");
```

Comment:
Just something to be aware of. This has a risk of lock congestion if `explict_bases` is very hot because we keep interning the same `TupleType`. I don't think `explict_bases` is that hot that this is an issue here but I thought it's worth noting (and it's also something that we might just need to fix in Salsa)